backend/rs/ic_vetkeys/src/utils/mod.rs

```diff
@@ -373,6 +373,91 @@ impl VetKey {
     }
 }
 
+/// Key material derived from a VetKey
+#[derive(Clone, Zeroize, ZeroizeOnDrop)]
+pub struct DerivedKeyMaterial {
+    key: Vec<u8>,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+/// An error while encrypting
+pub enum EncryptionFailed {
+    /// The provided message was too long to be encrypted
+    PlaintextTooLong,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+/// An error while decrypting
+pub enum DecryptionFailed {
+    /// The ciphertext was too short to possibly be valid
+    MessageTooShort,
+    /// The GCM tag did not validate
+    InvalidCiphertext,
+}
+
+impl DerivedKeyMaterial {
+    /// Create a new DerivedKeyMaterial
+    pub fn new(vetkey: &VetKey) -> Self {
+        Self {
+            key: vetkey.serialize().to_vec(),
+        }
+    }
+
+    fn derive_aes_gcm_key(&self, domain_sep: &str) -> Vec<u8> {
+        derive_symmetric_key(&self.key, domain_sep, 32)
```

Comment:
Do I understand correctly that the Rust implementation is compatible with the Typescript implementation regarding key-derivation/encryption/decryption, i.e., decrypting a ciphertext in the canister (with our Rust lib) that was encrypted in the browser (with our Typescript lib) would work? If so, if would make sense to explicitly mention this in the documentation.

Reply:
Exactly so

I'll update the docs